src/mmrelay/migrate.py

```
@@ -0,0 +1,1898 @@
+        if not dry_run:
+            _write_migration_state(
+                status="failed",
+                completed_steps=completed_steps,
+                error=str(exc),
+            )
+            rollback_result = rollback_migration(completed_steps=completed_steps)
+            report["rollback"] = rollback_result
+        return report
+    except (
+        Exception
+    ) as exc:  # noqa: BLE001 - catch-all for unexpected errors during migration
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (unused: `BLE001`)

Remove unused `noqa` directive

(RUF100)